.circleci/config.yml

```diff
@@ -46,21 +46,48 @@ jobs:
           key: klaatu-lint-{{ checksum "poetry.lock" }}
           paths:
             - .tox
-  test-nightly:
+  test-nightly-firefox:
     working_directory: ~/klaatu
     machine:
       image: ubuntu-2204:2024.01.1
+    environment:
+      FIREFOX_VERSION: -nightly
     steps:
       - checkout
       - run:
           name: Build docker image
-          command: docker build -t klaatu:latest -f Dockerfile.desktop .
+          command: docker compose up -d --build
+      - run:
+          name: Print firefox version
+          command: docker compose run -it klaatu bash -c "firefox --version"
       - run:
           name: Run tests
           command: |
-            set -ex
             docker compose up -d
             docker compose run klaatu tox -e bdd-tests -- --reruns 2 --experiment-branch control --variables tests/fixtures/test_experiment.json --run-update-test --private-browsing-enabled
+            echo $?
```

Comment:
Q: what does the `echo` do here (apart from log the exitCode)? Or I guess the question is "why"?

Reply:
Something with poetry and tox not reporting the correct error codes to circleci